zrml/prediction-markets/src/migrations.rs

```diff
@@ -1 +1,182 @@
+            prediction_markets_storage_version, PREDICTION_MARKETS_NEXT_STORAGE_VERSION,
+            "found unexpected prediction-markets pallet storage version. Found: {:?}. Expected: \
+             {:?}",
+            prediction_markets_storage_version, PREDICTION_MARKETS_NEXT_STORAGE_VERSION,
+        );
+
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::{mock::*, MomentOf};
+    use frame_support::assert_ok;
+    use orml_traits::MultiCurrency;
+    use zeitgeist_primitives::{
+        constants::{BASE, MILLISECS_PER_BLOCK},
+        types::{
+            Asset, BlockNumber, MarketCreation, MarketDispute, MarketDisputeMechanism,
+            MarketPeriod, MarketType, MultiHash, OutcomeReport, ScoringRule,
+        },
+    };
+
+    #[test]
+    fn test_on_runtime_upgrade_on_untouched_chain() {
+        ExtBuilder::default().build().execute_with(|| {
+            setup_chain();
+            RemoveDisputesOfResolvedMarkets::<Runtime>::on_runtime_upgrade();
+        });
+    }
+
+    #[test]
+    fn on_runtime_upgrade_updates_storage_versions() {
+        ExtBuilder::default().build().execute_with(|| {
+            setup_chain();
+            RemoveDisputesOfResolvedMarkets::<Runtime>::on_runtime_upgrade();
+            assert_eq!(
+                StorageVersion::get::<Pallet<Runtime>>(),
+                PREDICTION_MARKETS_NEXT_STORAGE_VERSION
+            );
+        });
+    }
+
+    #[test]
+    fn test_on_runtime_upgrade_with_sample_markets() {
+        ExtBuilder::default().build().execute_with(|| {
+            setup_chain();
+            let _ = Currency::deposit(Asset::Ztg, &ALICE, 1_000 * BASE);
+
+            // Markets which end here will have to be closed on migration:
+            let short_time: MomentOf<Runtime> = (5 * MILLISECS_PER_BLOCK).into();
+
+            create_test_market(
+                MarketPeriod::Timestamp(0..short_time),
+                MarketCreation::Permissionless,
+                MarketStatus::Resolved,
+            );
+
+            run_to_block(55);
+
+            // add one simple dispute for alreay resolved market to simulate a case
+            // where there is pendig dispute(s) for resolver market and that should
+            // be cleaned in storage migration.
+            let market_dispute =
+                MarketDispute { at: 55, by: CHARLIE, outcome: OutcomeReport::Categorical(0) };
+            let _res = crate::Disputes::<Runtime>::try_mutate(0, |disputes| {
+                disputes.try_push(market_dispute)
+            });
+
+            let disputes = crate::Disputes::<Runtime>::get(&0);
+            assert_eq!(disputes.len(), 1);
+
+            RemoveDisputesOfResolvedMarkets::<Runtime>::on_runtime_upgrade();
+
+            assert_eq!(MarketCommons::market(&0).unwrap().status, MarketStatus::Resolved);
+            let disputes = crate::Disputes::<Runtime>::get(&0);
+            assert_eq!(disputes.len(), 0);
+        });
+    }
```

Comment:
Maybe create at least two old markets with at least two disputes for each one of them.